### PR TITLE
add workflow to update nimbus experiments

### PR DESCRIPTION
**Pull request by jeddai**
This PR only affects the build process! After this is merged, it will run a workflow every 30 minutes to check for new first-run experiments for Fenix in experimenter. If there are some, it will write them to the `initial_experiments.json` file and open a PR with the changes. As soon as the changes are no longer necessary it will delete the PR and the branch.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Comment by Amejia481** [comment]
Awesome! Do we have some examples that we could use to test?

**Comment by jeddai** [comment]
So I had previously tested this by running the workflow against the experimenter staging API after creating a fenix experiment in experimenter's stage environment.

This run was used to test that it would successfully close out the PR if it was no longer necessary: 
https://github.com/mozilla-mobile/fenix/runs/7755453458?check_suite_focus=true

Here is an example of a run that resulted in a PR, and the PR to match:
https://github.com/jeddai/firefox-ios/runs/7774175380?check_suite_focus=true
https://github.com/jeddai/firefox-ios/pull/2

While the above is in my fork of firefox ios, it is almost the same workflow configuration (file path and PR label differences).

I'm happy to run any other tests that you'd like, we can re-run through my testing process for the fenix repo on a short call if you'd prefer.

**Review by Amejia481 — APPROVED** [review]
Thanks for the clarification. It looks good to me!
Is there a way for us to force the action to run?  I guess this will useful for when we don't have to wait for the job to run.

**Comment by Amejia481** [comment]
> If we want it to have the ability to be manually run I can make that change. Do you want me to add that?

Yeah, if it's not much work it could be really helpful!

**Comment by jeddai** [comment]
Done!

**Comment by Amejia481** [comment]
Thanks!!

**Comment by Amejia481** [comment]
I squashed the commit into a single one!

File: .github/workflows/update-nimbus-experiments.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          path: fenix
+          ref: main
+      - name: "Update Experiments JSON"
+        id: update-experiments-json
+        uses: mozilla-mobile/update-initial-experiments@v1
+        with:
+          repo-path: fenix
+          output-path: app/src/main/res/raw/initial_experiments.json
+          experimenter-url: https://experimenter.services.mozilla.com/api/v6/experiments/?is_first_run=true
+          app-name: fenix
```
Comment (jonalmeida) [inline .github/workflows/update-nimbus-experiments.yml:29]: Drive-by: what's the difference between app-name and repo-name?

Comment (jeddai) [reply .github/workflows/update-nimbus-experiments.yml:29]: The action's docs have more info: https://github.com/mozilla-mobile/update-initial-experiments

TL;DR: app-name is used to filter experiments downloaded from experimenter, repo-path is the relative or full path to the checked out repository within the workflow.

**Comment by Amejia481** [comment]
@Mergifyio refresh

**Comment by Amejia481** [comment]
@Mergifyio refresh
